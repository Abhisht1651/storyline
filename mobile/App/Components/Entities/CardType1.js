import React, { useState } from 'react'
import { View, Text, Image } from 'react-native'
import PropTypes from 'prop-types'
import { TouchableOpacity } from 'react-native-gesture-handler'

// Styles
import styles from '../Styles/EntitesStyles/CardType1Styles'
import images from '../../Themes/Images'
import { DefaultButton } from '../Buttons/DefaultButton'
import { X4SButton } from '../Buttons/X4SButton'

const CardType1 = ({ title, count, profileIcons }) => {
  const [pressStatus, setStatus] = useState(false)
  console.log(profileIcons)

  const changeStatus = () => {
    console.log('Clicked follow')
    setStatus(!pressStatus)
  }

  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={{uri:profileIcon}} key={index} style={styles.profileIcon} />
  ))

  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>
      <View style={styles.textOutline}>
        <View>
          <Text style={styles.textNameFont}>{title}</Text>
        </View>
        <View>
          <Text style={styles.textDetailsFont}>{`${count}K Following`}</Text>
        </View>
      </View>

      <View style={styles.followbuttonBox}>
        <TouchableOpacity onPress={changeStatus}>
          {!pressStatus ? (
            <DefaultButton outline title="Follow" />
          ) : (
            <X4SButton disabled title="Following" />
          )}
        </TouchableOpacity>
      </View>
    </View>
  )
}

CardType1.propTypes = {
  title: PropTypes.string,
  count: PropTypes.number,
  profileIcons: PropTypes.array
}

CardType1.defaultProps = {
  title: 'India',
  count: 0,
  profileIcons: [16]
}

export default CardType1
