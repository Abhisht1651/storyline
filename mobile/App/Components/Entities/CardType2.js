import React from 'react'
import { View, Text, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from '../Styles/EntitesStyles/CardType2Styles'
import images from '../../Themes/Images'

const CardType2 = ({ title, count, profileIcons }) => {
  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={profileIcon} key={index} style={styles.profileIcon} />
  ))
  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>

      <View style={styles.textOutline}>
        <View style={styles.textcenter}>
          <Text style={styles.textNameFont}>{title}</Text>
        </View>
        <View>
          <Text style={styles.textDetailsFont}>{`${count} friends`}</Text>
        </View>
        <View>
          <Text style={styles.textDetailsFont}>follow</Text>
        </View>
      </View>
    </View>
  )
}

CardType2.propTypes = {
  title: PropTypes.string,
  count: PropTypes.number,
  profileIcons: PropTypes.array
}

CardType2.defaultProps = {
  title: 'Bharat',
  count: 0,
  profileIcons: [images.EntitiesImage0]
}

export default CardType2
