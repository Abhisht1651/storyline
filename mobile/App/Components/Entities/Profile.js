import React from 'react'
import { View, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from '../Styles/EntitesStyles/ProfileStyles'
import images from '../../Themes/Images'

const Profile = ({ profileIcons }) => {
  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={profileIcon} key={index} style={styles.profileIcon} />
  ))
  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>
    </View>
  )
}

Profile.propTypes = {
  profileIcons: PropTypes.array
}

Profile.defaultProps = {
  profileIcons: [images.EntitiesImage0]
}

export default Profile
