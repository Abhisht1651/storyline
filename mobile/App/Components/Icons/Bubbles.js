import React from 'react'
import { View, TouchableOpacity } from 'react-native'

// Styles
import styles from '../Styles/IconStyles/BubblesStyles'
import RavenSVG from '../../Images/Icons/Path-1.svg'
import ShareSVG from '../../Images/Icons/Path-3.svg'
import ReactSVG from '../../Images/Icons/Path.svg'

const Bubbles = () => {
  return (
    <View style={styles.cardBox}>
      <TouchableOpacity>
        <View style={styles.profileIconBox}>
          <ShareSVG />
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.marginBox}>
        <View style={styles.profileIconBox}>
          <RavenSVG />
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.marginBox}>
        <View style={styles.profileIconBox}>
          <ReactSVG />
        </View>
      </TouchableOpacity>
    </View>
  )
}

export default Bubbles
