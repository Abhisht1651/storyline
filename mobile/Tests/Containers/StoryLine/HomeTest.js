// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import Home from '../../../App/Containers/StoryLine/Home'
import Bubbles from '../../../App/Components/Icons/Bubbles'
import Entities from '../../../App/Components/Entities/ThumbsLarge'
import DropDown from '../../../App/Components/Dropdowns/DropDown2'
import StoryLineShortCard from '../../../App/Components/StoryLineShortCardDefualt'
import RelatedPoll from '../../../App/Components/RelatedPoll'

describe('Home Screen', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find('Text')).toHaveLength(23)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find('Image')).toHaveLength(9)
  })
  it('should render the DefaultButton Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find('DefaultButton')).toHaveLength(1)
  })
  it('should render the Bubbles Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find(Bubbles)).toHaveLength(1)
  })
  it('should render the Entities Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find(Entities)).toHaveLength(6)
  })
  it('should render the DropDown Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find(DropDown)).toHaveLength(1)
  })
  it('should render the StoryLineShortCard Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find(StoryLineShortCard)).toHaveLength(3)
  })
  it('should render the RelatedPoll Component', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find(RelatedPoll)).toHaveLength(3)
  })
})
