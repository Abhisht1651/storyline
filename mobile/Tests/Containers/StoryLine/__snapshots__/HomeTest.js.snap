// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 67.5,
        "width": 750,
      }
    }
  />
  <ScrollViewMock>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 61,
            "justifyContent": "space-between",
            "width": 650,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 13.5,
            }
          }
        />
        <Bubbles />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "marginTop": 77,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1F1F60",
                  "fontFamily": "BananaGrotesk-Regular",
                  "fontSize": 14,
                  "letterSpacing": -0.08,
                  "opacity": 1,
                  "textAlign": "left",
                }
              }
            >
              World
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#1F1F60",
                  "borderRadius": 2,
                  "height": 4,
                  "marginHorizontal": 10,
                  "width": 4,
                }
              }
            />
            <Image
              source={
                Object {
                  "default": "",
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#1F1F60",
                  "fontFamily": "BananaGrotesk-Regular",
                  "fontSize": 14,
                  "letterSpacing": -0.08,
                  "opacity": 1,
                  "textAlign": "left",
                }
              }
            >
               Trending
            </Text>
          </View>
          <ForwardRef>
            <ForwardRef>
              <DefaultButton
                disabled={false}
                onPress={[Function]}
                outline={true}
                style={Object {}}
                title="Follow"
              />
            </ForwardRef>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 12,
                "letterSpacing": -0.07,
                "opacity": 1,
                "paddingLeft": 5,
                "paddingRight": 10,
                "textAlign": "left",
              }
            }
          >
            45
            K Reached
          </Text>
          <Image
            source={
              Object {
                "default": "",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 12,
                "letterSpacing": -0.07,
                "opacity": 1,
                "paddingLeft": 5,
                "paddingRight": 10,
                "textAlign": "left",
              }
            }
          >
            241K Followers
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 81.5,
              "width": 450,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 22,
                "letterSpacing": 0,
                "lineHeight": 29,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Brexit: UK exit from the European Union
          </Text>
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 12,
                "letterSpacing": -0.07,
                "marginTop": 5,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Updated 
            2m
             ago
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 80,
              "marginTop": 37.5,
              "width": 600,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Light",
                "fontSize": 14,
                "letterSpacing": 0,
                "lineHeight": 22,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            British sovereign bonds are rallying strongly after the EU warned over the weekend that Boris Johnson faces an uphill battle in striking a Brexit.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 41,
              "width": 649.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.12,
                "lineHeight": 54,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Related Entities
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.09,
                  "opacity": 1,
                  "textAlign": "right",
                }
              }
            >
              View all
            </Text>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <ScrollViewMock
            contentContainerStyle={
              Object {
                "paddingHorizontal": 48,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
              }
            }
            horizontal={true}
          >
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "marginBottom": 48,
                  "marginRight": 30,
                  "marginTop": 17,
                  "width": 150,
                }
              }
            >
              <ThumbsLarge
                profileIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                  ]
                }
                title="Default"
              />
            </View>
          </ScrollViewMock>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 86,
            "width": 650,
            "zIndex": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 40,
              "letterSpacing": -0.12,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          TimeLine
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginRight": 20,
              "width": 176,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 28,
                "letterSpacing": -0.12,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Sort by 
          </Text>
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Bold",
                "fontSize": 28,
                "letterSpacing": -0.12,
                "opacity": 1,
                "paddingRight": 5,
                "textAlign": "left",
              }
            }
          >
            Latest
          </Text>
          <DropDown2
            content={
              Array [
                "Latest Events First",
                "Oldest Events First",
              ]
            }
            getOption={[Function]}
            headerTitle="Sort by"
            type={2}
            zIndex={1}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 42,
            "marginTop": 78,
            "width": 650,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 6,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 38,
                "marginRight": 5,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 28,
                "letterSpacing": -0.12,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Major Events
          </Text>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 6,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 38,
                "marginRight": 5,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 28,
                "letterSpacing": -0.12,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            All events
          </Text>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "display": "none",
          }
        }
      >
        <Timeline
          circleColor="#47C3F4"
          circleSize={20}
          columnFormat="single-column-left"
          data={
            Array [
              Object {
                "dotColor": "47C3F4",
                "source": "Forbes",
                "storyDescription": "Ray Contreras talks about the different accents",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
              },
              Object {
                "source": "Forbes",
                "storyDescription": "Ray Contreras talks about the different accents",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
              },
              Object {
                "source": "Forbes",
                "storyDescription": "Ray Contreras talks about the different accents",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
              },
              Object {
                "source": "Forbes",
                "storyDescription": "Ray Contreras talks about the different accents",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
              },
            ]
          }
          dotSize={14}
          innerCircle="dot"
          isAllowFontScaling={true}
          lineColor="#47C3F4"
          lineWidth={2}
          renderDetail={[Function]}
          separator={false}
          showTime={false}
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F5F8F8",
            "justifyContent": "center",
            "paddingLeft": 22,
            "paddingTop": 72,
            "width": 750,
          }
        }
      >
        <Timeline
          circleColor="#47C3F4"
          circleSize={20}
          columnFormat="single-column-left"
          data={
            Array [
              Object {
                "comments": 34,
                "dotColor": "47C3F4",
                "onPressComments": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
                "topComment": Object {
                  "boostCount": 450,
                  "commentText": "When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself.",
                  "isReply": false,
                  "name": "Jammie Rice",
                  "replyCount": 1,
                  "selectionHandler": [Function],
                },
              },
              Object {
                "comments": 34,
                "onPressComments": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
                "topComment": Object {
                  "boostCount": 450,
                  "commentText": "When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself.",
                  "isReply": false,
                  "name": "Jammie Rice",
                  "replyCount": 1,
                  "selectionHandler": [Function],
                },
              },
              Object {
                "comments": 34,
                "onPressComments": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "storyImage": Object {
                  "default": "",
                },
                "time": "Sept 21, 2019, 5:00PM ET",
                "topComment": Object {
                  "boostCount": 450,
                  "commentText": "When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself.",
                  "isReply": false,
                  "name": "Jammie Rice",
                  "replyCount": 1,
                  "selectionHandler": [Function],
                },
              },
            ]
          }
          dotSize={14}
          innerCircle="dot"
          isAllowFontScaling={true}
          lineColor="#47C3F4"
          lineWidth={2}
          renderDetail={[Function]}
          separator={false}
          showTime={false}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 53.5,
            "minHeight": 267,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 66.5,
              "width": 649.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.12,
                "lineHeight": 54,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Related Poll
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.09,
                  "opacity": 1,
                  "textAlign": "right",
                }
              }
            >
              View all
            </Text>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <ScrollViewMock
            contentContainerStyle={
              Object {
                "paddingHorizontal": 48,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
              }
            }
            horizontal={true}
          >
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <relatedPoll
                border={true}
                isPollActive={true}
                onPressVote={[Function]}
                option={
                  Array [
                    "India",
                    "West Indies",
                    "Australia",
                    "England",
                  ]
                }
                question="Who are your favorites for the World Cup 2020?"
                timeLeft={2}
                voterIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                  ]
                }
                votes={54}
              />
            </View>
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <relatedPoll
                border={true}
                isPollActive={true}
                onPressVote={[Function]}
                option={
                  Array [
                    "India",
                    "West Indies",
                    "Australia",
                    "England",
                  ]
                }
                question="Who are your favorites for the World Cup 2020?"
                timeLeft={2}
                voterIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                  ]
                }
                votes={54}
              />
            </View>
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <relatedPoll
                border={true}
                isPollActive={true}
                onPressVote={[Function]}
                option={
                  Array [
                    "India",
                    "West Indies",
                    "Australia",
                    "England",
                  ]
                }
                question="Who are your favorites for the World Cup 2020?"
                timeLeft={2}
                voterIcons={
                  Array [
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                    Object {
                      "default": "",
                    },
                  ]
                }
                votes={54}
              />
            </View>
          </ScrollViewMock>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 53.5,
            "minHeight": 267,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 66.5,
              "width": 649.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.12,
                "lineHeight": 54,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Related StoryLine
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.09,
                  "opacity": 1,
                  "textAlign": "right",
                }
              }
            >
              View all
            </Text>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <ScrollViewMock
            contentContainerStyle={
              Object {
                "paddingHorizontal": 48,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
              }
            }
            horizontal={true}
          >
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <storyLineShortCardDefault
                border={true}
                followDisplay={true}
                followerDisplay={true}
                following={true}
                friendsFollowing="45 friends Following"
                heading="World"
                onPressFollow={[Function]}
                reach={45}
                storyImage={
                  Object {
                    "default": "",
                  }
                }
                storyTitle="UK exit from the European Union"
                trending={true}
                updatedTime="2m"
              />
            </View>
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <storyLineShortCardDefault
                border={true}
                followDisplay={true}
                followerDisplay={true}
                following={true}
                friendsFollowing="45 friends Following"
                heading="World"
                onPressFollow={[Function]}
                reach={45}
                storyImage={
                  Object {
                    "default": "",
                  }
                }
                storyTitle="UK exit from the European Union"
                trending={true}
                updatedTime="2m"
              />
            </View>
            <View
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <storyLineShortCardDefault
                border={true}
                followDisplay={true}
                followerDisplay={true}
                following={true}
                friendsFollowing="45 friends Following"
                heading="World"
                onPressFollow={[Function]}
                reach={45}
                storyImage={
                  Object {
                    "default": "",
                  }
                }
                storyTitle="UK exit from the European Union"
                trending={true}
                updatedTime="2m"
              />
            </View>
          </ScrollViewMock>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 53.5,
            "minHeight": 267,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 66.5,
              "width": 649.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.12,
                "lineHeight": 54,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Gallery (
            3
             Items)
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.09,
                  "opacity": 1,
                  "textAlign": "right",
                }
              }
            >
              View all
            </Text>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <ScrollViewMock
            contentContainerStyle={
              Object {
                "paddingHorizontal": 48,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
              }
            }
            horizontal={true}
          >
            <View
              key="0"
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  Object {
                    "default": "",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 356,
                    "width": 600,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#1F1F60",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 32,
                    "letterSpacing": -0.12,
                    "marginTop": 17,
                    "opacity": 1,
                    "textAlign": "left",
                  }
                }
              >
                Labor conference in London
              </Text>
            </View>
            <View
              key="1"
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  Object {
                    "default": "",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 356,
                    "width": 600,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#1F1F60",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 32,
                    "letterSpacing": -0.12,
                    "marginTop": 17,
                    "opacity": 1,
                    "textAlign": "left",
                  }
                }
              >
                Labor conference in London
              </Text>
            </View>
            <View
              key="2"
              style={
                Object {
                  "borderColor": "#E6ECED",
                  "borderRadius": 15,
                  "borderWidth": 0.5,
                  "height": 600,
                  "marginBottom": 133.5,
                  "marginRight": 15,
                  "marginTop": 17,
                  "opacity": 1,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  Object {
                    "default": "",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 356,
                    "width": 600,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#1F1F60",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 32,
                    "letterSpacing": -0.12,
                    "marginTop": 17,
                    "opacity": 1,
                    "textAlign": "left",
                  }
                }
              >
                Labor conference in London
              </Text>
            </View>
          </ScrollViewMock>
        </View>
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
